src/main/java/com/varukha/onlinebookstore/mapper/ShoppingCartMapper.java

```diff
@@ -0,0 +1,27 @@
+package com.varukha.onlinebookstore.mapper;
+
+import com.varukha.onlinebookstore.config.MapperConfig;
+import com.varukha.onlinebookstore.dto.shoppingcart.CreateShoppingCartRequestDto;
+import com.varukha.onlinebookstore.dto.shoppingcart.ShoppingCartDto;
+import com.varukha.onlinebookstore.model.CartItem;
+import com.varukha.onlinebookstore.model.ShoppingCart;
+import java.util.stream.Collectors;
+import org.mapstruct.AfterMapping;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.MappingTarget;
+
+@Mapper(config = MapperConfig.class, uses = CartItemMapper.class)
+public interface ShoppingCartMapper {
+
+    @Mapping(source = "shoppingCart.user.id", target = "userId")
+    ShoppingCartDto toDto(ShoppingCart shoppingCart);
+
+    @AfterMapping
+    default void setCartItems(@MappingTarget ShoppingCart shoppingCart,
+                              CreateShoppingCartRequestDto shoppingCartDto) {
+        shoppingCart.setCartItems(shoppingCartDto.getCartItems().stream()
+                .map(CartItem::new)
+                .collect(Collectors.toSet()));
```

Comment:
You can use Mapping annotation for it too, like you did with userId

Reply:
I don't know why, but it's work even without mapping)

Reply:
But for such purposes you actually need mapper, why do you want to keep it here?